/*
 * OncoKB APIs for Premium Users
 * These endpoints are designed for premium users. Please contact OncoKB team(contact@oncokb.org) if you want to be upgraded.
 *
 * OpenAPI spec version: v1.0.0
 * Contact: contact@oncokb.org
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package org.oncokb.client;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.IOException;

/**
 * AnnotatedVariant
 */
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2020-02-06T11:50:04.572-05:00")
public class AnnotatedVariant {
  @SerializedName("entrezGeneId")
  private Integer entrezGeneId = null;

  @SerializedName("gene")
  private String gene = null;

  @SerializedName("isoform")
  private String isoform = null;

  @SerializedName("mutationEffect")
  private String mutationEffect = null;

  @SerializedName("mutationEffectAbstracts")
  private String mutationEffectAbstracts = null;

  @SerializedName("mutationEffectPmids")
  private String mutationEffectPmids = null;

  @SerializedName("oncogenicity")
  private String oncogenicity = null;

  @SerializedName("proteinChange")
  private String proteinChange = null;

  @SerializedName("refSeq")
  private String refSeq = null;

  @SerializedName("variant")
  private String variant = null;

  public AnnotatedVariant entrezGeneId(Integer entrezGeneId) {
    this.entrezGeneId = entrezGeneId;
    return this;
  }

   /**
   * Get entrezGeneId
   * @return entrezGeneId
  **/
  @ApiModelProperty(value = "")
  public Integer getEntrezGeneId() {
    return entrezGeneId;
  }

  public void setEntrezGeneId(Integer entrezGeneId) {
    this.entrezGeneId = entrezGeneId;
  }

  public AnnotatedVariant gene(String gene) {
    this.gene = gene;
    return this;
  }

   /**
   * Get gene
   * @return gene
  **/
  @ApiModelProperty(value = "")
  public String getGene() {
    return gene;
  }

  public void setGene(String gene) {
    this.gene = gene;
  }

  public AnnotatedVariant isoform(String isoform) {
    this.isoform = isoform;
    return this;
  }

   /**
   * Get isoform
   * @return isoform
  **/
  @ApiModelProperty(value = "")
  public String getIsoform() {
    return isoform;
  }

  public void setIsoform(String isoform) {
    this.isoform = isoform;
  }

  public AnnotatedVariant mutationEffect(String mutationEffect) {
    this.mutationEffect = mutationEffect;
    return this;
  }

   /**
   * Get mutationEffect
   * @return mutationEffect
  **/
  @ApiModelProperty(value = "")
  public String getMutationEffect() {
    return mutationEffect;
  }

  public void setMutationEffect(String mutationEffect) {
    this.mutationEffect = mutationEffect;
  }

  public AnnotatedVariant mutationEffectAbstracts(String mutationEffectAbstracts) {
    this.mutationEffectAbstracts = mutationEffectAbstracts;
    return this;
  }

   /**
   * Get mutationEffectAbstracts
   * @return mutationEffectAbstracts
  **/
  @ApiModelProperty(value = "")
  public String getMutationEffectAbstracts() {
    return mutationEffectAbstracts;
  }

  public void setMutationEffectAbstracts(String mutationEffectAbstracts) {
    this.mutationEffectAbstracts = mutationEffectAbstracts;
  }

  public AnnotatedVariant mutationEffectPmids(String mutationEffectPmids) {
    this.mutationEffectPmids = mutationEffectPmids;
    return this;
  }

   /**
   * Get mutationEffectPmids
   * @return mutationEffectPmids
  **/
  @ApiModelProperty(value = "")
  public String getMutationEffectPmids() {
    return mutationEffectPmids;
  }

  public void setMutationEffectPmids(String mutationEffectPmids) {
    this.mutationEffectPmids = mutationEffectPmids;
  }

  public AnnotatedVariant oncogenicity(String oncogenicity) {
    this.oncogenicity = oncogenicity;
    return this;
  }

   /**
   * Get oncogenicity
   * @return oncogenicity
  **/
  @ApiModelProperty(value = "")
  public String getOncogenicity() {
    return oncogenicity;
  }

  public void setOncogenicity(String oncogenicity) {
    this.oncogenicity = oncogenicity;
  }

  public AnnotatedVariant proteinChange(String proteinChange) {
    this.proteinChange = proteinChange;
    return this;
  }

   /**
   * Get proteinChange
   * @return proteinChange
  **/
  @ApiModelProperty(value = "")
  public String getProteinChange() {
    return proteinChange;
  }

  public void setProteinChange(String proteinChange) {
    this.proteinChange = proteinChange;
  }

  public AnnotatedVariant refSeq(String refSeq) {
    this.refSeq = refSeq;
    return this;
  }

   /**
   * Get refSeq
   * @return refSeq
  **/
  @ApiModelProperty(value = "")
  public String getRefSeq() {
    return refSeq;
  }

  public void setRefSeq(String refSeq) {
    this.refSeq = refSeq;
  }

  public AnnotatedVariant variant(String variant) {
    this.variant = variant;
    return this;
  }

   /**
   * Get variant
   * @return variant
  **/
  @ApiModelProperty(value = "")
  public String getVariant() {
    return variant;
  }

  public void setVariant(String variant) {
    this.variant = variant;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AnnotatedVariant annotatedVariant = (AnnotatedVariant) o;
    return Objects.equals(this.entrezGeneId, annotatedVariant.entrezGeneId) &&
        Objects.equals(this.gene, annotatedVariant.gene) &&
        Objects.equals(this.isoform, annotatedVariant.isoform) &&
        Objects.equals(this.mutationEffect, annotatedVariant.mutationEffect) &&
        Objects.equals(this.mutationEffectAbstracts, annotatedVariant.mutationEffectAbstracts) &&
        Objects.equals(this.mutationEffectPmids, annotatedVariant.mutationEffectPmids) &&
        Objects.equals(this.oncogenicity, annotatedVariant.oncogenicity) &&
        Objects.equals(this.proteinChange, annotatedVariant.proteinChange) &&
        Objects.equals(this.refSeq, annotatedVariant.refSeq) &&
        Objects.equals(this.variant, annotatedVariant.variant);
  }

  @Override
  public int hashCode() {
    return Objects.hash(entrezGeneId, gene, isoform, mutationEffect, mutationEffectAbstracts, mutationEffectPmids, oncogenicity, proteinChange, refSeq, variant);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AnnotatedVariant {\n");
    
    sb.append("    entrezGeneId: ").append(toIndentedString(entrezGeneId)).append("\n");
    sb.append("    gene: ").append(toIndentedString(gene)).append("\n");
    sb.append("    isoform: ").append(toIndentedString(isoform)).append("\n");
    sb.append("    mutationEffect: ").append(toIndentedString(mutationEffect)).append("\n");
    sb.append("    mutationEffectAbstracts: ").append(toIndentedString(mutationEffectAbstracts)).append("\n");
    sb.append("    mutationEffectPmids: ").append(toIndentedString(mutationEffectPmids)).append("\n");
    sb.append("    oncogenicity: ").append(toIndentedString(oncogenicity)).append("\n");
    sb.append("    proteinChange: ").append(toIndentedString(proteinChange)).append("\n");
    sb.append("    refSeq: ").append(toIndentedString(refSeq)).append("\n");
    sb.append("    variant: ").append(toIndentedString(variant)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

